app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,51 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      # Temporarily only supporting atomic.
+      raise 'Service Type composite not supported' if data['service_type'] == 'composite'
+      create_atomic(data).tap(&:save!)
+    rescue => err
+      raise BadRequestError, "Could not create Service Template - #{err}"
+    end
+
+    private
+
+    def create_atomic(data)
+      service_template = ServiceTemplate.new(data.except('request_info'))
+      if data.key?('request_info')
+        service_template.add_resource(create_service_template_request(data['request_info']))
```

Comment:
I think you need to make `data['request_info']` with more structures instead of a plain hash. You code does not create service_template properly for `generic_orchestration`, `generic_ansible_tower`, and `generic_container`(to come). For these types different entries are added to service_resources table.